### PR TITLE
Feat/phil frontend

### PR DESCRIPTION
**Pull request by PhilPinsdorf**
Added the Frontend for the Random Generator

### CODE REVIEW

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+          v-if="generatorState.state === 'stopped'"
+          class="flex h-screen items-center justify-center"
+        >
+          <Avatar
+            class="scale-[130%]"
+            :src="generatorState.user?.avatarUrl"
+            :firstname="generatorState.user?.firstname"
+            :lastname="generatorState.user?.lastname"
+          />
+          <audio autoplay>
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:63]: Only the part that is animated by the transition should be here.

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+          </div>
+        </div>
+      </Transition>
+
+      <Transition name="winner">
+        <div
+          v-if="generatorState.state === 'stopped'"
+          class="flex h-screen items-center justify-center"
+        >
+          <Avatar
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:57]: Components must NOT consist of one word. Names of components MUST consist of several words. See: [Multi-word component names](https://v2.vuejs.org/v2/style-guide/?redirect=true#Multi-word-component-names-essential). In this case `CardAvatar` (below in the comments on the file I suggest another file path). If you are interessted in how this works see: https://github.com/fsr5-fhaachen/portals/blob/main/vite.config.js#L37-L41

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            class="absolute right-0 top-1/2 h-2/3 -translate-y-1/2 scale-x-[-1] transform"
+            src="/images/random-generator/gifs/trumpet.gif"
+          />
+        </div>
+      </Transition>
+    </div>
+  </div>
+</template>
+
+<style>
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:83]: The single-file component top-level element order is either script > template > style or, as in our case, template > script > style. See: https://v2.vuejs.org/v2/style-guide/?redirect=true#Single-file-component-top-level-element-order-recommended

File: routes/web.php [hunk 4]
```diff
@@ -84,6 +84,7 @@
         ], function () {
             Route::get('/random-generator', [DashboardAdminRandomGeneratorController::class, 'index'])->name('dashboard.admin.randomGenerator.index');
             Route::post('/random-generator', [DashboardAdminRandomGeneratorController::class, 'indexExecuteSubmit'])->name('dashboard.admin.randomGenerator.indexExecuteSubmit');
+            Route::get('/random-generator/display', [DashboardAdminRandomGeneratorController::class, 'display'])->name('');
```
Comment (TitusKirch) [inline routes/web.php:87]: Add the missing name

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/components/card/Avatar.vue [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<template>
```
Comment (TitusKirch) [inline resources/js/components/card/Avatar.vue:1]: This component is special for the RandomGenerator and should therefore be in `resources/js/components/random/generator/user/AvatarCard.vue`, which makes the name `RandomGeneratorUserAvatarCard`.

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+// locale state variables
+const generatorState = ref<{
+  state: "setup" | "idle" | "running" | "stopped";
+  user?: Models.User;
+}>({
+  state: "setup",
+});
+
+const isFetchingGenerator = ref(false);
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:136]: We should name this `isFetchingRandomGenerator`.

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+const props = defineProps({
+  users: {
+    type: Array as PropType<App.Models.User[]>,
+    required: true,
+  },
+});
+
+// locale state variables
+const generatorState = ref<{
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:129]: We should name this `randomGeneratorState`.

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+      props.users.sort(() => Math.random() - 0.5);
+    }
+
+    generatorState.value = data;
+  }
+
+  isFetchingGenerator.value = false;
+};
+
+const generatorInterval = setInterval(fetchRandomGeneratorState, 500);
```
Comment (TitusKirch) [inline resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue:167]: We should name this `randomGeneratorInterval`.

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue (hunk 1 above)
Comment (PhilPinsdorf) [reply resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue]: Fixed, just put audio somewhere else

File: resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue (hunk 2 above)
Comment (PhilPinsdorf) [reply resources/js/pages/Dashboard/Admin/RandomGenerator/Display.vue]: Used the name from later in this thread

File: resources/js/components/random/generator/user/AvatarCard.vue [hunk 9]
```diff
@@ -5,10 +5,9 @@
     <img v-if="props.src" :src="props.src" class="h-fit" />
     <div
       v-else
-      class="flex h-[30rem] transform items-center justify-center text-center font-['EIGHTY_MILES'] text-8xl text-white"
+      class="flex h-[30rem] transform items-center justify-center text-center font-eighty-miles text-8xl text-white"
     >
-      {{ props.firstname }} <br />
-      {{ props.lastname }}
+      <span>{{ props.firstname }} {{ props.lastname }}</span>
```
Comment (TitusKirch) [inline resources/js/components/random/generator/user/AvatarCard.vue:10]: I meant first and last name each in their own span

File: resources/js/types/components.d.ts [hunk 10]
```diff
@@ -12,7 +12,7 @@ declare module "vue" {
     AppMessage: typeof import("./../components/app/Message.vue")["default"];
     AppNavbar: typeof import("./../components/app/Navbar.vue")["default"];
     BoxContainer: typeof import("./../components/box/Container.vue")["default"];
-    CardAvatar: typeof import("./../components/card/Avatar.vue")["default"];
+    CardAvatarCard: typeof import("./../components/card/AvatarCard.vue")["default"];
```
Comment (TitusKirch) [inline resources/js/types/components.d.ts:15]: Did you regenerate the file or why is the line still there? Could you test with `npm run dev` and then lint via. `npm run lint:fix
./vendor/bin/pint`

**Review by TitusKirch — CHANGES_REQUESTED** [review]

File: resources/js/components/random/generator/user/AvatarCard.vue (hunk 9 above)
Comment (PhilPinsdorf) [reply resources/js/components/random/generator/user/AvatarCard.vue:10]: That doesn't work unfortunately
This approach works. Is it too long it wraps around, else it stays in one line

Comment (TitusKirch) [reply resources/js/components/random/generator/user/AvatarCard.vue:10]: but this approach is not good. And look, the other version will work 
image

I will do this than in my branch
image

Comment (PhilPinsdorf) [reply resources/js/components/random/generator/user/AvatarCard.vue:10]: Hö lol I did that and it didnt look like that
But then I will put it in

File: resources/js/types/components.d.ts (hunk 10 above)
Comment (TitusKirch) [reply resources/js/types/components.d.ts:15]: I merge you code in my branch, and so this good fixed automatically

File: resources/js/components/random/generator/user/AvatarCard.vue (hunk 9 above)
Comment (PhilPinsdorf) [reply resources/js/components/random/generator/user/AvatarCard.vue:10]: Maybe, just maybe, I was too stupid to save

Comment (TitusKirch) [reply resources/js/components/random/generator/user/AvatarCard.vue:10]: Have you added the `flex-col` class?

**Review by TitusKirch — APPROVED** [review]

**Pull request merged**